version: '3.9'
services:
  db:
    container_name: mypostgres
    image: postgres
    volumes:
      - './postgres-data:/var/lib/postgresql/data'
    ports:
      - '5433:5432'
    network_mode: bridge
    environment:
      - POSTGRES_PASSWORD=root
      - POSTGRES_USER=fazztrack
    expose:
      - 5432
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U fazztrack" ]
      interval: 10s
      timeout: 5s
      retries: 5
  backend:
    container_name: coffeebackapp
    image: sasmeka/coffeeback
    ports:
      - '8081:8080'
    expose:
      - 8080
    # networks:
    #   - local
    network_mode: bridge
    environment:
      - PGHOST=host.docker.internal
      - PGPORT=5433
      - PGUSER=fazztrack
      - PGDATABASE=fazztrack
      - PGPASSWORD=root
    depends_on:
      db:
        condition: service_healthy
    # networks:
    #   local:
    #     name: 'bridge'
    #     external: true

    # docker run --name coffeebackapp --net=bridge -e PGHOST=host.docker.internal -e PGPORT=5433 -e PGUSER=fazztrack -e PGDATABASE=postgres -e PGPASSWORD=root -p 8081:8080 sasmeka/coffeeback
    # docker run -d -e POSTGRES_PASSWORD=root -e POSTGRES_USER=fazztrack -p 5433:5432 -v ./postgres-data:/var/lib/postgresql/data --name mypostgres postgres
    # docker compose up/down
